packages/shared/src/components/cards/v1/FeedItemContainer.tsx

```diff
@@ -0,0 +1,90 @@
+import React, {
+  AnchorHTMLAttributes,
+  forwardRef,
+  HTMLAttributes,
+  ReactElement,
+  ReactNode,
+  Ref,
+  useState,
+} from 'react';
+import Link from 'next/link';
+import classNames from 'classnames';
+import { Post } from '../../../graphql/posts';
+import { Card, CardLink } from './Card';
+import { RaisedLabel, RaisedLabelType } from './RaisedLabel';
+import { useFeedPreviewMode } from '../../../hooks';
+import { TypeLabel } from './TypeLabel';
+import ConditionalWrapper from '../../ConditionalWrapper';
+
+interface FeedItemContainerProps {
+  flagProps?: FlagProps;
+  children: ReactNode;
+  domProps: HTMLAttributes<HTMLDivElement>;
+  linkProps?: AnchorHTMLAttributes<HTMLAnchorElement>;
+}
+
+interface FlagProps extends Pick<Post, 'pinnedAt' | 'trending' | 'type'> {
+  adAttribution?: ReactElement | string;
+}
+
+function FeedItemContainer(
+  { flagProps, children, domProps, linkProps }: FeedItemContainerProps,
+  ref?: Ref<HTMLElement>,
+): ReactElement {
+  const { adAttribution, pinnedAt, trending, type } = flagProps;
+
+  const raisedLabelType = pinnedAt
+    ? RaisedLabelType.Pinned
+    : RaisedLabelType.Hot;
+  const description =
+    raisedLabelType === RaisedLabelType.Hot && trending > 0
+      ? `${trending} devs read it last hour`
+      : undefined;
+  const isFeedPreview = useFeedPreviewMode();
+  const showFlag = (!!pinnedAt || !!trending) && !isFeedPreview && description;
+  const showTypeLabel = !!adAttribution || !!type;
+  const [focus, setFocus] = useState(false);
+
+  return (
+    <Card
+      {...domProps}
+      data-testid="postItem"
+      ref={ref}
+      className={classNames(domProps?.className, focus && 'bg-theme-float')}
+    >
+      {linkProps && (
+        <ConditionalWrapper
+          condition={!linkProps.href.startsWith('https://')}
```

Comment:
Maybe we can use our existing function of `isValidHttpUrl` if you are trying to ensure only valid URLs to be passed.

Reply:
Yeah this conditional render feels a bit odd, also could it not be handled inside `CardLink`?

Reply:
This condition is there so that we do not render a nextjs `<Link>` for links that point outside of our app. E.g. if it's a global link, we just want the `<a>` element there. It's a bit naive, but don't know if we have a better solution for that somewhere already.

Reply:
I see, there are places where links are with http or https for internal pages, this is because it was built to support the extension. Since this is for the feed card's wrapper, it will always be pointing to internal links when clicking the card.

Reply:
I just checked and nextjs already has this check as part of the `<Link` component, so I'll remove the conditional rendering. Should make it simpler hopefully :)